[docs] Add a guide for setting dark mode by default (#34839)
mui/material-ui: docs/data/joy/guides/applying-dark-mode/applying-dark-mode.md

  setPreBodyComponents([getInitColorSchemeScript()]);
}
```

## Dark mode by default

> Set `defaultMode: 'dark'` to use dark color scheme for the first time users.

Reviewer: ```suggestion
To have dark mode as the default theme of your app, set `defaultMode: 'dark'`. 
```

Small copy iteration.
